packages/cli/src/executionLifecycleHooks/saveExecutionProgress.ts

```diff
@@ -0,0 +1,105 @@
+import { Container } from 'typedi';
+
+import type { IRunExecutionData, ITaskData, IWorkflowBase } from 'n8n-workflow';
+import { ErrorReporterProxy as ErrorReporter } from 'n8n-workflow';
+
+import { Logger } from '@/Logger';
+import { ExecutionRepository } from '@/databases/repositories/execution.repository';
+import { toSaveSettings } from '@/executionLifecycleHooks/toSaveSettings';
+
+export async function saveExecutionProgress(
```

Comment:
The reason why I prefer creating `@Service` classes is because then we can inject dependencies (like `Logger`, and `ExecutionRepository`) and avoid doing `Container.get`.

Another advantage of that is then we can write unit tests injecting mocks into the service constructor, without requiring `typedi` in the tests.

Reply:
Totally agree!

We've been working with Omar and Micha on breaking down the execution lifecycle into standalone utilities at `cli/src/executionLifecycleHooks` as a stepping stone to finally refactoring the hooks into classes that are easier to reason about, so this is an interim state.

I think it might be too early to start creating service classes out of the utilities before this effort is completed, but please let me know if you think we should tackle this now.